### ISSUE TITLE
Locale Issues

### ISSUE THREAD

**Issue by blitzmann** [pyfa-org/Pyfa#1010]
**Note:** if you're having issues with pyfa finding your path, please create a separate issue for it and then post here with the issue # so that I can ad it to the list below. I would like to keep discussions contained to their own threads.

---

I'll be consolidating all locale issues here and progress to fix them.

A bit of background: A fix was submitted by @Ebag333 (#910) and put into place for v1.27.0 which seemingly fixed some locale issues that we were having by coming up with our own path joining system that automatically handled unicode paths. When v1.27.0 released, reports sprung up from folks who were previously able to use pyfa who could no longer do so (specifically from Russian locales). 

After researching the issue myself, I can't find much information on unicode issues inherent with `os.path.join` (different from `walk` which might have some issues, but we don't use it). Additionally, through multiple testing on Windows and Linux with Unicode paths and joining them in different ways, I could only get #548 to reproduce (never got around to testing OS X, but it should function in much the same ), which itself is an easy fix (see 6e643d75). 

I decided to roll back the changes to path joining introduced for v1.27.0 so that we can get 1.27.1 out the door for those that need it, and will be evaluating any potential locale issues on a case-by-case bases moving forward. From what I've researched, there shouldn't be a reason we can't handle unicode paths with the standard `os.path.join` - if there's an issue with it, we probably aren't en/decoding something correctly and should focus specifically on that as it comes up. If we need the code again, we can simply revert b5835898, but I would prefer to take a scalpel to these issues as they pop up. 

Path issues can be particularly tricky to get information for since we can't even really do logs with it (we can't be sure the path to log to is correct). To counteract this, I'll be developing a simple wxPython window which will pop up instead of the MainFrame whenever pyfa configuration throws an exception. This should allow us to display the error, along with whatever values we are using and want to report on (the path that we're trying to set, etc).

### Related Issues

(probably missed some old, closed issues, but whatever).

* #379 Hebrew Locale issues (waiting on @WiseFerret to confirmation if this is still an issue with latest master)
* ~~#548~~ Fixed with 6e643d75
* ~~#997~~ Introduced with v1.27.0, should be fixed due to the revert of #910

**Comment by blitzmann** [comment]
@Ebag333 as you've been working a lot on this issue, I have a favor to ask:

Can you list out simply and clearly your test cases? I'm having trouble keeping up with know bad combinations (hence why I started this issue)

For example:

<table>
<tr><th>Path</th><th>Platform</th><th>System Locale</th><th>Result</th></td></tr>
<tr><td>`c:\Users\Me\some russian path\pyfa\`</td><td>Windows</td><td>Russian (cp-whatever)</td><td>Works</td></td></tr>
<tr><td>`same as above`</td><td>Windows</td><td>English</td><td>Fails (provide trace)</td></td></tr>
</table>

(and whatever other data you think might prove useful)

**Comment by WiseFerret** [comment]
Can you post a link to latest master for linux so I can test it? I'm not yet familiar with git.

**Comment by blitzmann** [comment]
@WiseFerret Check Slack, I uploaded a file for you :)

**Comment by Ebag333** [comment]
>Can you list out simply and clearly your test cases? I'm having trouble keeping up with known bad combinations (hence why I started this issue)

That's why I spend a good chunk of last evening writing out tests in a PR that still isn't merged. :)

**Comment by blitzmann** [comment]
I want it in addition to the tests. Tests are only as reliable and accurate as it's programmed to be, and there could be mistakes that we don't see, unless you're suggestion your code is infallible :) I would like to have manual verification and documentation for at least some test cases first and a simple up/down result (with trace for failed). The problem is that right now we have these tests and the results strewn about in various threads, comments, and paragraphs. 

For example, that large reply you made to the closed PR had some results, but to extract what was tested, in what environment, with which codec, and the result is difficult because it's written through conversation. I simply would like it consolidated here.

Thanks :)

**Comment by Ebag333** [comment]
I understand.

The goal of that test PR is to flesh out those examples so you can replicate them yourself.  Without that it's not worth much.  Give me some time, I'm trying to fix the *OTHER* stuff you're complaining about . ;)

**Comment by blitzmann** [comment]
Closing this as locale issue tracking, if any, have gone out the window with pyfa 2.x. I'm hoping python 3 is now able to understand various locales, but if you run into issues with this on 2.x, please open a new issue. Thanks! 😄
